cmake/partition_manager.cmake

```diff
@@ -74,6 +74,24 @@ if (NOT static_configuration AND CONFIG_PM_IMAGE_NOT_BUILT_FROM_SOURCE)
     place the image correctly in flash.")
 endif()
 
+# From Documentation: By default, the Partition Manager dynamically
+# places the partitions in memory. However, if you have a deployed
+# product that consists of multiple images, where only a subset of
+# the included images can be upgraded through a firmware update
+# mechanism, the upgradable images must be statically configured.
+if (NOT static_configuration AND
+  (CONFIG_BOOTLOADER_MCUBOOT OR CONFIG_SECURE_BOOT))
+      message(WARNING "
+        ---------------------------------------------------------------------
+        --- WARNING: Using a bootloader without pm_static.yml.            ---
+        --- Static partitioning is required for non-upgradable partitions.---
+        --- See Partition Manager docs on Static Configuration for more   ---
+        --- info.                                                         ---
+        ---------------------------------------------------------------------
+        \n"
+      )
```

Comment:
Comment and message are opposed, and message is actually wrong

Reply:
Or possibly `Static partitioning is required when using a non-upgradable bootloader.` ?

Reply:
I like the last sentence

Reply:
`Static partitioning is required when using a non-upgradable bootloader.`

Is confusing, it would be better to just link to or quote the documentation as it is on https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/scripts/partition_manager/partition_manager.html#static-configuration

Reply:
Cant find a way to include the link without exceeding line-length.
So I try to copy the text from docs in.
Then the warning will look like this in a build:
![image](https://github.com/nrfconnect/sdk-nrf/assets/89006246/caf536e3-0d97-4a7d-bc9e-08ad1384b0cd)
What do you think about this?

Reply:
That looks great!